src/DoctrineWriter.php

```diff
@@ -60,57 +47,68 @@ class DoctrineWriter implements Writer, Writer\FlushableWriter
     protected $truncate = true;
 
     /**
-     * List of fields used to lookup an object
-     *
-     * @var array
+     * @var LookupStrategy
      */
-    protected $lookupFields = [];
+    private $lookupStrategy;
 
     /**
-     * Method used for looking up the item
+     * Create a Doctrine writer with an object lookup strategy.
+     *
+     * @param string         $objectName
+     * @param ObjectManager  $objectManager
+     * @param LookupStrategy $lookupStrategy
      *
-     * @var array
+     * @return self
      */
-    protected $lookupMethod;
+    public static function withLookupStrategy(
+        $objectName,
+        ObjectManager $objectManager,
+        LookupStrategy $lookupStrategy
+    ) {
+        return new self($objectManager, $objectName, null, $lookupStrategy);
+    }
 
     /**
      * Constructor
      *
-     * @param ObjectManager $objectManager
-     * @param string        $objectName
-     * @param string|array  $index         Field or fields to find current entities by
-     * @param string        $lookupMethod  Method used for looking up the item
+     * @param ObjectManager  $objectManager
+     * @param string         $objectName
+     * @param string|array   $index          Field or fields to find current
+     *                                       entities by
+     * @param string         $lookupMethod   Method used for looking up the item
+     * @param LookupStrategy $lookupStrategy
+     *
+     * @throws UnsupportedDatabaseTypeException
      */
     public function __construct(
         ObjectManager $objectManager,
         $objectName,
```

Comment:
`$objectName` needs to passed both here and when constructing a `LookupStrategy`. Not very elegant, but I don’t see any other options now.